core/settings.py

```diff
@@ -136,6 +136,15 @@
 # Password validation
 # https://docs.djangoproject.com/en/4.1/ref/settings/#auth-password-validators
 
+# Password hashing using bcrypt
+PASSWORD_HASHERS = [
+    'django.contrib.auth.hashers.BCryptSHA256PasswordHasher',  # Built-in bcrypt with SHA256
+    'django.contrib.auth.hashers.Argon2PasswordHasher',
+    'django.contrib.auth.hashers.PBKDF2PasswordHasher',
+    'django.contrib.auth.hashers.PBKDF2SHA1PasswordHasher',
+]
```

Comment:
can you provide some justification for using bcrypt over the django framework default of pbkdf2?
also BCryptSHA256PasswordHasher fails to load due to a missing library

Reply:
The reason for using BCryp (To enhance password security): 
1. Stronger Defense Against Hardware Cracking
bcrypt has a design that is less susceptible to hardware-accelerated attacks, such as those leveraging GPUs or ASICs, because of its reliance on a memory-hard key derivation process. PBKDF2 lacks memory-hardness, making it easier to parallelize on modern hardware.
2. Built-in Salting Mechanism
bcrypt automatically includes a salt when hashing, ensuring that even identical passwords result in completely different hashes. PBKDF2 also supports salting, but it requires developers to manage it correctly, increasing the risk of implementation errors.
3. Memory Usage
bcrypt has a lower memory requirement compared to some PBKDF2 implementations, making it more efficient in resource-constrained environments.

Reply:
Evidence of BCrypt working:
<img width="712" alt="Screenshot 2024-12-04 at 14 48 23" src="https://github.com/user-attachments/assets/e24e3f20-a530-45eb-be81-e293336bfe3c">